docs/developer-guides/migration-guide-1.4.0.md

```diff
@@ -0,0 +1,114 @@
+
+### Runtime upgrade to v1.3.1
+
+This patch version extends the Sidechain pallet with a new extrinsic `upgrade_and_set_addresses`
+which allows the on-chain governance to atomically upgrade the runtime and set the genesis utxo and new main chain
+scripts to observe.
+
+1. Update Partner Chains SDK dependencies in your `Cargo.toml` to `v1.3.1`
+2. Add the following configuration to your runtime configuration:
+```rust
+impl pallet_sidechain::Config for Runtime {
+    // ... other parameters
+	type MainChainScripts = sp_session_validator_management::MainChainScripts;
+
+	fn set_main_chain_scripts(scripts: Self::MainChainScripts) {
+		pallet_session_validator_management::MainChainScriptsConfiguration::<Runtime>::set(scripts);
+	}
+}
+```
+This will allow the `upgrade_and_set_addresses` extrinsic to update configuration of the SessionValidatorManagement
+pallet together with the Runtime code.
+3. Increment the `spec_version` in your runtime configuration.
+4. Build the new Runtime WASM (you can do it by running `cargo build --release`)
+5. Upgrade the runtime of the chain using the newly built WASM (it can be found in `$CARGO_TARGET_DIR/release/wbuild/<runtime name>/<runtime name>.compressed.wasm`) and the
+extrinsic `system/setCode`.
+
+After these steps the chain should be running the runtime using Partner Chains SDK v1.3.1. This version is
+fully backwards-compatible and is a preparatory step towards v1.4.0
+
+### Release new version using PC-SDK v1.4.0
+
+This version of the node is backwards-compatible with the runtime in versions 1.3.x. and is needed to support
+the runtime in version v1.4.0.
+
+1. Update all Partner Chains SDK dependencies in your `Cargo.toml` file to `v1.4.0`
+2. Modify your runtime and node code to be compatible with PC-SDK version v1.4.0. Consult the release notes for v1.4.0
+for required changes.
+3. Release a new version of your node. This step depends on your release process.
+4. Upgrade nodes running the chain to the newly released version. This step depends on your deployment process.
+
+After this step, the network should be composed of nodes v1.4.0 but the runtime would remain at v1.3.1, and be ready for the matching
+runtime upgrade to v1.4.0.
+
+### New main chain state setup
+
+Because of the changes to the smart contracts and their on-chain data, a completely new set
+of main chain data needs to be set up on Cardano.
+
+1. Pick the initial governance authority.
+This can be a newly generated or already existing main chain key set controlled by the Partner Chain authority.
+2. Run the `prepare-configuration` command of `partner-chains-cli` (v1.4.0) in a fresh directory.
+This step will involve selecting the _genesis UTXO_ to be spent intializing the governance mechanism.
+Save the `partner-chains-cli-chain-config.json` file produced by this step.
+3. Run the `setup-main-chain-state` command of `partner-chains-cli`, setting up the D-param and permissioned candidates.
```

Comment:
`create-chain-spec` step is mandatory. Because `setup-main-chain-state` runs partner-chains-node with `--chain chain-spec.json`

Reply:
Actually no. This chain-spec can be sent to SPOs as well. But it is important to know, that it won't be used as genesis of any chain, only used by the tooling.

Reply:
I added that